qiskit_machine_learning/primitives/kernels/quantum_kernel.py

```diff
@@ -0,0 +1,162 @@
+                left_parameters, right_parameters, shape
+            )
+
+        else:
+            left_parameters, right_parameters = self._get_parametrization(x_vec, y_vec)
+            kernel_matrix = self._get_kernel_matrix(left_parameters, right_parameters, shape)
+
+        if is_symmetric and self._enforce_psd:
+            kernel_matrix = self._make_psd(kernel_matrix)
+        return kernel_matrix
+
+    def _get_parametrization(self, x_vec: np.ndarray, y_vec: np.ndarray) -> Tuple[np.ndarray]:
+        """
+        Combines x_vec and y_vec to get all the combinations needed to evaluate the kernel entries.
+        """
+        x_count = x_vec.shape[0]
+        y_count = y_vec.shape[0]
+
+        left_parameters = np.zeros((x_count * y_count, x_vec.shape[1]))
+        right_parameters = np.zeros((x_count * y_count, y_vec.shape[1]))
+        index = 0
+        for x_i in x_vec:
+            for y_j in y_vec:
+                left_parameters[index, :] = x_i
+                right_parameters[index, :] = y_j
+                index += 1
+
+        return left_parameters, right_parameters
+
+    def _get_symmetric_parametrization(self, x_vec: np.ndarray) -> Tuple[np.ndarray]:
+        """
+        Combines two copies of x_vec to get all the combinations needed to evaluate the kernel entries.
+        """
+        x_count = x_vec.shape[0]
+
+        left_parameters = np.zeros((x_count * (x_count + 1) // 2, x_vec.shape[1]))
+        right_parameters = np.zeros((x_count * (x_count + 1) // 2, x_vec.shape[1]))
+
+        index = 0
+        for i, x_i in enumerate(x_vec):
+            for x_j in x_vec[i:]:
+                left_parameters[index, :] = x_i
+                right_parameters[index, :] = x_j
+                index += 1
+
+        return left_parameters, right_parameters
+
+    def _get_kernel_matrix(
+        self, left_parameters: np.ndarray, right_parameters: np.ndarray, shape: Tuple
+    ) -> np.ndarray:
+        """
+        Given a parametrization, this computes the symmetric kernel matrix.
+        """
+        kernel_entries = self._fidelity.compute(left_parameters, right_parameters)
+        kernel_matrix = np.zeros(shape)
+
+        index = 0
+        for i in range(shape[0]):
+            for j in range(shape[1]):
+                kernel_matrix[i, j] = kernel_entries[index]
+                index += 1
+        return kernel_matrix
+
+    def _get_symmetric_kernel_matrix(
+        self, left_parameters: np.ndarray, right_parameters: np.ndarray, shape: Tuple
+    ) -> np.ndarray:
+        """
+        Given a set of parametrization, this computes the kernel matrix.
+        """
+        kernel_entries = self._fidelity.compute(left_parameters, right_parameters)
+        kernel_matrix = np.zeros(shape)
+        index = 0
+        for i in range(shape[0]):
+            for j in range(i, shape[1]):
+                kernel_matrix[i, j] = kernel_entries[index]
+                index += 1
```

Comment:
Code duplication with the previous method. Could parametrize the starting point of the inner `for` loop - `0` vs `i`.